opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchScrollRequest.java

```diff
@@ -143,15 +147,50 @@ public SearchScrollRequest scrollRequest() {
    * to be reused across different physical plan.
    */
   public void reset() {
-    scrollId = null;
+    scrollId = NO_SCROLL_ID;
   }
 
   /**
    * Convert a scroll request to string that can be included in a cursor.
    * @return a string representing the scroll request.
    */
   @Override
-  public String toCursor() {
-    return scrollId;
+  public boolean hasAnotherBatch() {
+    return !needClean && !scrollId.equals(NO_SCROLL_ID);
+  }
+
+  @Override
+  public void writeTo(StreamOutput out) throws IOException {
+    initialSearchRequest.writeTo(out);
+    out.writeTimeValue(scrollTimeout);
+    out.writeBoolean(needClean);
+    if (!needClean) {
+      // If needClean is true, there is no more data to get from OpenSearch and scrollId is
+      // used only to clean up OpenSearch context.
+
+      out.writeString(scrollId);
+    }
```

Comment:
I guess you need to throw an `NoCursorException` in `else`

Reply:
Throwing by `OpenSearchIndexScan` is sufficient.

Reply:
True, I missed that.